server/isomorphic-routing/index.js

```diff
@@ -37,6 +39,15 @@ export function serverRouter( expressApp, setUpRoute, section ) {
 	};
 }
 
+function setRouteMiddleware( context, next ) {
+	context.store.dispatch( setRouteAction(
+		context.pathname,
+		context.query
+	) );
+
+	next();
```

Comment:
It's not a redux middleware though but an express.js one and don't use the value returned.